lib/ansible/plugins/inventory/foreman.py

```diff
@@ -230,6 +233,12 @@ def _populate(self):
                 if self.get_option('want_facts'):
                     self.inventory.set_variable(host['name'], 'ansible_facts', self._get_facts(host))
 
+                hostvars = combine_vars(get_group_vars(self.inventory.hosts[host['name']].get_groups()), self.inventory.hosts[host['name']].get_vars())
+                self._set_composite_vars(self.get_option('compose'), hostvars, host['name'], strict=True)
```

Comment:
If a user provides the `strict` option in the constructed doc fragment it is disregarded since it's hardcoded to True. The following two lines could be updated with that as well.
```suggestion
                self._set_composite_vars(self.get_option('compose'), hostvars, host['name'], strict=self.get_option('strict'))
```